### PR TITLE
Update artifact provenance documentation

### PR DESCRIPTION
**Pull request by jhrozek**
Fixes: #2388

### CODE REVIEW

**Comment by jhrozek** [comment]
@rdimitrov PTAL

File: docs/docs/how-to/artifact_signatures.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+```
+minder profile create -f profile-artifact-provenance.yaml
+```
+
+Once the profile is created, Minder will start checking the artifacts produced
+by the enrolled repositories and the policy status will be updated
+accordingly. If the artifact is not matching the expected provenance (for
+example someone pushes a new image to the registry after having signed the
+image with their personal account), a violation is presented via the profile
+status.
```
Comment (rdimitrov) [inline docs/docs/how-to/artifact_signatures.md:123]: Should we mention the use case where the artifact was produced from another workflow that is not the expected one? Also the same comment about the potential alert we'll open.

Comment (jhrozek) [reply docs/docs/how-to/artifact_signatures.md]: done!

File: docs/docs/how-to/artifact_signatures.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+artifact:
+  - type: artifact_signature
+    params:
+      tags: [latest]
+      name: good-repo-go
+    def:
+      is_signed: true
+      is_verified: true
+      branch: main
+      workflow_name: build-image-signed-ghat.yml
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply docs/docs/how-to/artifact_signatures.md]: fixed

File: docs/docs/how-to/artifact_signatures.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+Create the profile in Minder:
+```
+minder profile create -f profile-artifact-simple.yaml
+```
+
+Once the profile is created, Minder will start checking the artifacts produced by the enrolled repositories
+and the policy status will be updated accordingly. If the artifact is not matching the expected provenance
+(for example someone pushes a new image to the registry without signing it), a
+violation is presented via the profile status.
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply docs/docs/how-to/artifact_signatures.md]: good idea, added

File: docs/docs/how-to/artifact_signatures.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+```
+git clone https://github.com/stacklok/minder-rules-and-profiles.git
+```
+
+In that directory you can find all the reference rules and profiles.
+```
+cd minder-rules-and-profiles
+```
+
+Create the `pr_vulnerability_check` rule type in Minder:
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply docs/docs/how-to/artifact_signatures.md]: oops, copy-pasta

**Review by rdimitrov — APPROVED** [review]

**Pull request merged**